src/parser/ParsedQuery.cpp

```diff
@@ -290,6 +290,24 @@ void ParsedQuery::addSolutionModifiers(SolutionModifiers modifiers) {
       // TODO<qup42, joka921> Check that all variables used in the expression of
       //  Aliases are visible in the QueryBody.
     }
+  } else if (hasConstructClause()) {
+    if (_groupByVariables.empty()) {
+      return;
+    }
+
+    for (const auto& triple : constructClause()) {
+      for (const auto& varOrTerm : triple) {
+        if (auto variable = std::get_if<Variable>(&varOrTerm)) {
+          if (!ad_utility::contains(_groupByVariables, *variable)) {
+            throw ParseException("Variable " + variable->name() +
+                                 " is used but not "
+                                 "aggregated despite the query not being "
+                                 "grouped by " +
+                                 variable->name() + ".");
+          }
+        }
```

Comment:
Doesn't the same error message appear somewhere else when we check for the SELECT clause? 
I think then both checks should be here (We have the invariant then, that the SELECT/CONSTRUCT clause must be set before the solution modifiers, but that is something we want to enforce anyway, and these lines should be a function/lambda that factors out the common code.